### PR TITLE
Fix possible wrong database being assigned in post-taxprofiling steps

### PR DESCRIPTION
**Pull request by jfy133**
Status:

 - [x] Fix taxpasta standardise being executed instead of `merge` 

---

resolves https://github.com/nf-core/taxprofiler/issues/299

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
# TODO

- [ ] Test with multiple databases
- [ ] Implement for all other standalone 'report' modules in `profiling.nf`
- [ ] Implement for all other modules with dbs in `standardisation_profiles.nf`

**Comment by jfy133** [comment]
Problem: Taxpasta merge/standardise sin't working, only standardise is being executed, but `merge` is expected... 

Possibly database channel is being consumed for the first sample when doing the join prior multiMap before the profiling tool bit...?

**Comment by jfy133** [comment]
Just need to check MOTUS_MERGE gets executed correctly and then GTG!

**Comment by jfy133** [comment]
Confirmed!

**Review by LilyAnderssonLee — COMMENTED** [review]
I guess the KrakenUniq test will pass now , right?

**Review by LilyAnderssonLee — APPROVED** [review]

File: subworkflows/local/standardisation_profiles.nf [hunk 1]
```diff
@@ -117,7 +117,19 @@ workflow STANDARDISATION_PROFILES {
                                     [[id:it[0]], it[1]]
                                 }
 
-    KAIJU_KAIJU2TABLE_COMBINED ( ch_profiles_for_kaiju, ch_input_databases.kaiju.map{it[1]}, params.kaiju_taxon_rank)
+    ch_input_for_kaiju2tablecombine = ch_profiles_for_kaiju
+                                        .dump(tag: "B41")
```
Comment (Midnighter) [inline subworkflows/local/standardisation_profiles.nf:121]:
```suggestion
```

File: subworkflows/local/standardisation_profiles.nf [hunk 2]
```diff
@@ -117,7 +117,19 @@ workflow STANDARDISATION_PROFILES {
...
+                                        .map { meta, profile -> [meta['id'], meta, profile] }
+                                        .dump(tag: "B42")
```
Comment (Midnighter) [inline subworkflows/local/standardisation_profiles.nf:123]:
```suggestion
```

File: subworkflows/local/standardisation_profiles.nf [hunk 3]
```diff
@@ -117,7 +117,19 @@ workflow STANDARDISATION_PROFILES {
...
+                                        .combine(ch_input_databases.kaiju.map{meta, db -> [meta.db_name, meta, db]}, by: 0)
```
Comment (Midnighter) [inline subworkflows/local/standardisation_profiles.nf:124]: I prefer the `.key` way to the `['key']` of accessing map values myself, but I feel it should be done consistently.

File: subworkflows/local/standardisation_profiles.nf [hunk 4]
```diff
@@ -117,7 +117,19 @@ workflow STANDARDISATION_PROFILES {
...
+                                        .dump(tag: "AFTER")
```
Comment (Midnighter) [inline subworkflows/local/standardisation_profiles.nf:125]:
```suggestion
```

**Review by Midnighter — CHANGES_REQUESTED** [review]
Couple of minor comments. In general this looks fine to me.

We might consider introducing a helper function that encapsulates the `map -> combine -> multiMap` pattern.

File: subworkflows/local/standardisation_profiles.nf (hunk 3 above)
Comment (jfy133) [reply subworkflows/local/standardisation_profiles.nf:122]: Agreed! Have gone with the shorter version :+1:

**Comment by jfy133** [comment]
> We might consider introducing a helper function that encapsulates the map -> combine -> multiMap pattern.

Agreed, I tried this initially but couldn't get it to work properly. We will be doing a few refactoring things (As you've brought up other functions in the past too), but this is quite a big booboo so rather get a fix in sooner rather than later

**Review by Midnighter — APPROVED** [review]

**Pull request merged**